src/main/java/net/datafaker/transformations/JsonTransformer.java

```diff
@@ -7,149 +7,185 @@
+            sb.append(generate(Arrays.asList((Object[]) value)));
+        } else {
+            String val = String.valueOf(value);
+            boolean toWrap = !val.startsWith("#{json");
+            if (toWrap) {
+                sb.append("\"");
+            }
+            for (char c : String.valueOf(value).toCharArray()) {
+                sb.append(ESCAPING_MAP.getOrDefault(c, c + ""));
+            }
+            if (toWrap) {
+                sb.append("\"");
+            }
+        }
+    }
+
+    private static String generate(Collection<Object> collection) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("[");
+        int i = 0;
+        for (Object value : collection) {
+            if (i > 0) {
+                sb.append(", ");
+            }
+            i++;
+            value2String(value, sb);
+        }
+        sb.append("]");
+        return sb.toString();
+    }
+
+
+    private static Map<Character, String> createEscapeMap() {
+        final Map<Character, String> map = new HashMap<>();
+        map.put('\\', "\\\\");
+        map.put('\"', "\\\"");
+        map.put('\b', "\\b");
+        map.put('\f', "\\f");
+        map.put('\n', "\\n");
+        map.put('\r', "\\r");
+        map.put('\t', "\\t");
+        map.put('/', "\\/");
+        map.put('\u0000', "\\u0000");
+        map.put('\u0001', "\\u0001");
+        map.put('\u0002', "\\u0002");
+        map.put('\u0003', "\\u0003");
+        map.put('\u0004', "\\u0004");
+        map.put('\u0005', "\\u0005");
+        map.put('\u0006', "\\u0006");
+        map.put('\u0007', "\\u0007");
+        // map.put('\u0008', "\\u0008");
+        // covered by map.put('\b', "\\b");
+        // map.put('\u0009', "\\u0009");
+        // covered by map.put('\t', "\\t");
+        // map.put((char) 10, "\\u000A");
+        // covered by map.put('\n', "\\n");
+        map.put('\u000B', "\\u000B");
+        // map.put('\u000C', "\\u000C");
+        // covered by map.put('\f', "\\f");
+        // map.put((char) 13, "\\u000D");
+        // covered by map.put('\r', "\\r");
+        map.put('\u000E', "\\u000E");
+        map.put('\u000F', "\\u000F");
+        map.put('\u0010', "\\u0010");
+        map.put('\u0011', "\\u0011");
+        map.put('\u0012', "\\u0012");
+        map.put('\u0013', "\\u0013");
+        map.put('\u0014', "\\u0014");
+        map.put('\u0015', "\\u0015");
+        map.put('\u0016', "\\u0016");
+        map.put('\u0017', "\\u0017");
+        map.put('\u0018', "\\u0018");
+        map.put('\u0019', "\\u0019");
+        map.put('\u001A', "\\u001A");
+        map.put('\u001B', "\\u001B");
+        map.put('\u001C', "\\u001C");
+        map.put('\u001D', "\\u001D");
+        map.put('\u001E', "\\u001E");
+        map.put('\u001F', "\\u001F");
+        return Collections.unmodifiableMap(map);
+    }
+    private static char[] delimitedBy(boolean formatAsArray) {
+        return formatAsArray ? new char[]{'[',']'} : new char[] {'{','}'};
```

Comment:
It seems it creates a new array each call which is not good for performance.
There are cases when we generate millions of rows and this will slow this down